script/gen_requirements_all.py

```diff
@@ -279,8 +279,15 @@ def requirements_test_all_output(reqs):
         if any(
             # Always install requirements that are not part of integrations
             not mdl.startswith("homeassistant.components.") or
-            # Install tests for integrations that have tests
-            has_tests(mdl)
+            # Install requirements for integrations that have tests
+            has_tests(mdl) or
+            # ...or whose dependencies have tests
+            any(
```

Comment:
I don't think this check is correct. You don't care about it's dependencies, you care about integrations that depend on this one.

Reply:
FTR, I have noted this but haven't had time to revisit the issue. Would be very much appreciated if someone knows right off the bat how this would be done best would chime in and just implement and push it to my branch -- this is the test suite everyone's supposed to be running after all and it's been broken for various reason for a long time here and there (including this issue), not some obscure thing only I'm using and pushing to be included :)